File: Java/SharedPreferencesTest/app/src/main/java/com/example/sharedpreferencestest/MainActivity.java
package com.example.sharedpreferencestest;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText etId;
    private EditText etPw;
    private Button btLogin;
    private TextView tvId;
    private TextView tvPw;
    private Button btPop;

    private SharedPreferences preferences;
    private SharedPreferences.Editor editor;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etId = findViewById(R.id.etId);
        etPw = findViewById(R.id.etPw);
        btLogin = findViewById(R.id.btLogin);
        tvId = findViewById(R.id.tvId);
        tvPw = findViewById(R.id.tvPw);
        btPop = findViewById(R.id.btPop);

        preferences = getSharedPreferences("UserInfo", MODE_PRIVATE);

        editor = preferences.edit();

        btLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                editor.putString("Id", etId.getText().toString()).commit();
                editor.putString("Pw", etPw.getText().toString()).commit();

                tvId.setText(preferences.getString("Id", ""));
                tvPw.setText(preferences.getString("Pw", ""));
            }
        });

        AlertDialog.Builder builder = new AlertDialog.Builder(this);


        btPop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                builder.setTitle("제목");
                builder.setMessage("메시지");
                
                builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        Toast.makeText(MainActivity.this, "Yes", Toast.LENGTH_SHORT).show();
                    }
                });
                
                builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        Toast.makeText(MainActivity.this, "No", Toast.LENGTH_SHORT).show();
                    }
                });
                
                AlertDialog dialog = builder.create();
                dialog.show();
            }
        });

    }
}

File: Kotlin/MutableLiveData/app/src/main/java/com/example/mutablelivedata/MainActivity.kt
package com.example.mutablelivedata

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import com.example.mutablelivedata.R

class MainActivity : AppCompatActivity() {
    private lateinit var button1 : Button
    private lateinit var button2 : Button
    private lateinit var textView : TextView
    private lateinit var button3 : Button
    private lateinit var button4 : Button
    private lateinit var textView2 : TextView

    private var liveData : MutableLiveData<String> = MutableLiveData()
    private var liveData2 : MutableLiveData<String> = MutableLiveData()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button1 = findViewById(R.id.button1)
        button2 = findViewById(R.id.button2)
        textView = findViewById(R.id.textView)

        textView2 = findViewById(R.id.textView2)
        button3 = findViewById(R.id.button3)
        button4 = findViewById(R.id.button4)

        liveData.observe(this, Observer{
            textView.text = it
        })

        liveData2.observe(this, Observer {
            textView2.text = it
        })

        button1.setOnClickListener {
            liveData.value = "Hello world!"
        }

        button2.setOnClickListener {
            liveData.value = "Hi hell"
        }

        button3.setOnClickListener {
            liveData2.value = "Hello world!"
        }

        button4.setOnClickListener {
            liveData2.value = "Hi hell"
        }

    }
}